core/src/main/java/org/apache/spark/memory/MemoryMode.java

```diff
@@ -22,5 +22,6 @@
 @Private
 public enum MemoryMode {
   ON_HEAP,
-  OFF_HEAP
+  OFF_HEAP,
+  ON_HEAP_CACHEDBATCH
```

Comment:
Current implementation relies on memory mode to allocate a kind of `ColumnVector`.
If we do not add a new memory model, I think that we have to introduce additional conditional branches in getter/setter.